apps/storefront/src/pages/OrderDetail/index.test.tsx

```diff
@@ -1215,4 +1253,806 @@ describe('when a personal customer visits an order', () => {
+
+      const dialog = await screen.findByRole('dialog', { name: 'Re-Order' });
+
+      expect(
+        within(dialog).getByText('Select products and quantity for reorder'),
+      ).toBeInTheDocument();
+
+      const productGroup = within(dialog).getByRole('group', { name: 'Laugh Canister' });
+
+      await userEvent.click(within(productGroup).getByRole('checkbox'));
+
+      await userEvent.click(await screen.findByRole('button', { name: 'Add to cart' }));
+
+      await waitFor(() => {
+        expect(screen.getByText('Products are added to cart')).toBeInTheDocument();
+      });
+
+      expect(window.b2b.callbacks.dispatchEvent).toHaveBeenCalledWith('on-cart-created', {
+        cartId: 'foo-bar',
+      });
+    });
+
+    it('can adjust the quantity of a product', async () => {
+      const screamCanister = buildProductWith({
+        name: 'Scream Canister',
+        quantity: 1,
+        product_options: [],
+      });
+
+      const createCartSimple = vi.fn();
+
+      server.use(
+        graphql.query('GetCustomerOrderStatuses', () =>
+          HttpResponse.json(buildCustomerOrderStatusesWith('WHATEVER_VALUES')),
+        ),
+        graphql.query('AddressConfig', () =>
+          HttpResponse.json(buildAddressConfigResponseWith('WHATEVER_VALUES')),
+        ),
+        graphql.query('GetCustomerOrder', () =>
+          HttpResponse.json(
+            buildCustomerOrderResponseWith({
+              data: { customerOrder: { products: [screamCanister] } },
+            }),
+          ),
+        ),
+        graphql.query('getCart', () => HttpResponse.json({ data: { site: { cart: null } } })),
+        graphql.mutation('createCartSimple', ({ variables }) =>
+          HttpResponse.json(createCartSimple(variables)),
+        ),
+      );
+
+      when(createCartSimple)
+        .calledWith({
+          createCartInput: {
+            lineItems: [
+              {
+                quantity: 12,
+                productEntityId: screamCanister.product_id,
+                variantEntityId: screamCanister.variant_id,
+                selectedOptions: { multipleChoices: [], textFields: [] },
+              },
+            ],
+          },
+        })
+        .thenReturn({ data: { cart: { createCart: { cart: { entityId: 'foo-bar' } } } } });
+
+      renderWithProviders(<OrderDetails />, { preloadedState });
+
+      await waitForElementToBeRemoved(() => screen.queryAllByRole('progressbar'));
+
+      await userEvent.click(screen.getByRole('button', { name: 'Re-Order' }));
+
+      const dialog = await screen.findByRole('dialog', { name: 'Re-Order' });
+
+      expect(
+        within(dialog).getByText('Select products and quantity for reorder'),
+      ).toBeInTheDocument();
+
+      const productGroup = within(dialog).getByRole('group', { name: 'Scream Canister' });
+
+      await userEvent.click(within(productGroup).getByRole('checkbox'));
+
+      await userEvent.type(within(productGroup).getByRole('spinbutton'), '2');
```

Comment:
might be worth a comment to explain we're appending `2` to the `1` to make `12` 🍹